### PR TITLE
feat: add `FeOffset` filter

### PR DESCRIPTION
**Pull request by jakex7**
# Summary

Continuation of #2316
Introducing new filter `FeOffset`.

## Test Plan

Example app -> Filters -> FeOffset

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added documentation in `README.md`
- [x] I updated the typed files (typescript)

### CODE REVIEW

**Review by WoLewicki — APPROVED** [review]
Left some comments, if they do not require change, then it should be good to go!

**Comment by WoLewicki** [comment]
Also, should we add this filter to `FilterImage` ? I can see such changes in PR with `FeGaussianBlur`

File: apple/Filters/RNSVGFilterPrimitive.mm [hunk 1]
```diff
@@ -93,6 +93,13 @@ - (CIImage *)applyFilter:(NSMutableDictionary<NSString *, CIImage *> *)results p
   return previous;
 }
 
+- (CIImage *)applyFilter:(NSMutableDictionary<NSString *, CIImage *> *)results
+    previousFilterResult:(CIImage *)previous
+                     ctm:(CGAffineTransform)ctm
+{
+  return [self applyFilter:results previousFilterResult:previous];
```
[the remark this answers is not in the document]
Comment (jakex7) [reply apple/Filters/RNSVGFilterPrimitive.mm:100]: Many of the filters do not require `CTM` so they can override either `applyFilter previousFilterResult ctm` or simply `applyFilter previousFilterResult`

File: src/lib/extract/extractFilter.ts [hunk 2]
```diff
@@ -49,7 +51,27 @@ export const extractFeColorMatrix = (
...
+  return extracted;
+};
+
+export const extractFeOffset = (
+  props: FeOffsetComponentProps
+): FeOffsetNativeProps => {
+  const extracted: FeOffsetNativeProps = {};
+
+  if (props.in) {
+    extracted.in1 = props.in;
```
[the remark this answers is not in the document]
Comment (jakex7) [reply src/lib/extract/extractFilter.ts:67]: It's described in the W3 specification, to not use reserved word

**Comment by jakex7** [comment]
> Also, should we add this filter to `FilterImage` ? I can see such changes in PR with `FeGaussianBlur`

It has been removed in https://github.com/software-mansion/react-native-svg/pull/2359 and added types in https://github.com/software-mansion/react-native-svg/pull/2361/commits/3a9a2884bfb5266a186665168e07a8da4899697a

**Pull request merged**